RMtools_3D/do_fitIcube.py

```diff
@@ -163,128 +217,249 @@ def make_model_I(fitsI, freqFile, polyOrd=2, cutoff=-1, prefixOut="",
         medTmp = np.nanmedian(dataPlane[idxSky])
         
         # Pass 2: use a fixed 3-sigma cutoff to mask off emission
-        idxSky = np.where(dataPlane < medTmp + rmsTmp * 3)
-        medSky = np.nanmedian(dataPlane[idxSky])
+        
+        idxSky = np.where(dataPlane < medTmp + rmsTmp * threshold)
+        medSky[i] = np.nanmedian(dataPlane[idxSky])
         rmsArr[i] = MAD(dataPlane[idxSky])
-        mskSky[idxSky] +=1
+
         
         # When building final emission mask treat +ve cutoffs as absolute
         # values and negative cutoffs as sigma values
-        if cutoff>0:
+        if cutoff > 0:
             idxSrc = np.where(dataPlane > cutoff)
         else:
-            idxSrc = np.where(dataPlane > medSky -1 * rmsArr[i] * cutoff)
+            idxSrc = np.where(dataPlane > medSky[i] -1 * rmsArr[i] * cutoff)
+
         mskSrc[idxSrc] +=1
 
-        # Clean up
+    end = time.time()
+    print(' For loop masking takes %.3fs'%(end-start))
+    return rmsArr, mskSrc
 
-    # Save the noise spectrum
-    if outDir == '':
-        outDir='.'
-    print("Saving the RMS noise spectrum in an ASCII file:")
-    outFile = outDir + "/"  + prefixOut + "Inoise.dat"
-    print("> %s" % outFile)
-    np.savetxt(outFile, rmsArr)
+    
+    
+def channel_noise(chan, datacube, header, cutoff, threshold, nBits):
```

Comment:
What is this function, and why is it distinct from `cube_noise`? It needs a docstring. Also, doesn't seem to be used anywhere?